core/resolver.go

```diff
@@ -236,3 +225,33 @@ func MoveSource(recipe *Recipe, source Source) error {
 		return fmt.Errorf("unsupported source type %s", source.Type)
 	}
 }
+
+// checksumValidation validates the checksum of a file
+func checksumValidation(source Source, path string) error {
+	//No checksum provided
+	if source.Checksum == "" {
+		return nil
+	}
+	//open and read the Readme.md file
+	file, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	//Calculate the checksum
+	checksum := sha256.New()
+	_, err = io.Copy(checksum, file)
+	if err != nil {
+		return fmt.Errorf("could not calculate tar file checksum")
+	}
+
+	fmt.Printf("%x", checksum.Sum(nil))
+	//Compare the checksums
+	fmt.Printf("%x", checksum.Sum(nil))
+	if fmt.Sprintf("%x", checksum.Sum(nil)) != source.Checksum {
```

Comment:
maybe it would make more sense to have `checksum.Sum` stored in a variable so that it does not always have to be calculated.

Reply:
Where debubing outputs sorry